src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useState } from 'react';
+import { Dispatch, SetStateAction } from 'react';
+
+export const TodoModal: React.FC<{
+  todos: Todo[];
+  selectedTodoId: number;
+  onClose: Dispatch<SetStateAction<number | null>>;
+}> = ({ todos, selectedTodoId, onClose }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [user, setUser] = useState<User | null>(null);
+  const todo = todos.find(innertodo => innertodo.id === selectedTodoId);
+  const { id, title, userId, completed } = todo as Todo;
+
+  function handleClose() {
+    onClose(null);
+  }
+
+  useEffect(() => {
+    getUser(userId).then(respons => {
```

Comment:
There is a typo in the variable name `respons`. It should be `response` to correctly reflect the data being received from the `getUser` function.